tests/test_curve.py

```diff
@@ -0,0 +1,159 @@
+
+    def curve_tester(self, curve_class, args):
+        curve = curve_class(*args)
+
+        # Serialization round trip
+        table = curve.toTable()
+        curve2 = curve_class.fromTable(table)
+        self.assertEqual(curve, curve2)
+
+        # via FITS
+        with lsst.utils.tests.getTempFilePath(".fits") as tmpFile:
+            curve.writeFits(tmpFile)
+            curve2 = algorithms.Curve.readFits(tmpFile)
+
+        self.assertEqual(curve2, curve)
+
+        # via text file
+        with lsst.utils.tests.getTempFilePath(".ecsv") as tmpFile:
+            curve.writeText(tmpFile)
+            curve2 = algorithms.Curve.readText(tmpFile)
+
+        self.assertEqual(curve2, curve)
+
+        # Check bad values
+        with self.assertRaises(ValueError):
+            # test that raised when non quantities are passed
+            nargs = []
+            for arg in args:
+                if hasattr(arg, 'unit'):
+                    nargs.append(arg.value)
+                else:
+                    nargs.append(arg)
+            _ = curve_class(*nargs)
+
+            nargs = []
+            for arg in args:
+                if hasattr(arg, 'unit'):
+                    nargs.append(arg._set_unit(None))
+                else:
+                    nargs.append(arg)
+            _ = curve_class(*nargs)
+
+    def interp_tester(self, curve_class, args, detector):
+        curve = curve_class(*args)
+        w = 3500*u.angstrom
+        xs = np.linspace(0, 1023, 33)
+        ys = np.linspace(0, 1023, 33)
+        val_map = {'A': 0.9329662, 'B': 0.7463730}
+        # Does interpolation work
+        for x, y in zip(xs, ys):
+            point = Point2D(x, y)
+            if detector:
+                amp = cgUtils.findAmp(detector, Point2I(point))
+                value = val_map[amp.getName()]
+            else:
+                value = 0.9329662
+            interp_val = curve.evaluate(detector, point, w)
+            self.assertAlmostEqual(interp_val.value, value, places=5)
+            self.assertEqual(interp_val.unit, u.percent)
+        # Does interpolation work with arrays
+        w_arr = np.linspace(320, 430, 70)*u.nm
+        out_arr = curve.evaluate(detector, point, w_arr)
+        self.assertEqual(len(w_arr), len(out_arr))
+        # Does interpolation with different units work as expected
+        point = Point2D(500., 500.)
+        val1 = curve.evaluate(detector, point, w)
+        new_w = w.to(u.mm)
+        val2 = curve.evaluate(detector, point, new_w)
+        self.assertEqual(val1.value, val2.value)
+        # interpolation with non-quantity should raise
+        with self.assertRaises(ValueError):
+            interp_val = curve.evaluate(detector, point, w.value)
+        # Does out of band interpolation do something reasonable
+        with self.assertRaises(ValueError):
+            w = 0.*u.angstrom
+            interp_val = curve.evaluate(detector, point, w)
+
+    def test_curve(self):
```

Comment:
It's not obvious to me that you are testing both Curve subclasses.

Reply:
Is this just a comment about the name of the class or a request to make it more obvious that I'm actually constructing both classes in this test?

Reply:
When I first looked at the test I couldn't work out if all Curve types were being tested. Can you add a docstring and a couple of explanatory comments to the test code?

Reply:
I ended up refactoring these into two distinct tests.  I think it's much clearer now.